Add immersive mode for posts view

dfsm opened the pull request:
Implements a new immersive reading experience that automatically hides the navigation bar and toolbar when scrolling down, and shows them when scrolling up.

  Features:
  • Toggle available in main settings and posts settings
  • Velocity-based transitions (0.5 threshold)
  • Automatic bar reveal when near top/bottom of content
  • Gradient overlay for improved status bar readability
  • Seamless light/dark theme integration
  • Smooth 0.25s animations with proper layout updates

  Files modified:
  • AwfulSettings: Added immersionModeEnabled setting
  • AwfulSettingsUI: Added settings toggles and localization
  • PostsPageView: Core scroll handling and UI hiding logic
  • PostsPageViewController: Navigation bar integration
  • PostsPageSettingsViewController: In-page settings toggle

nolanw commented on App/View Controllers/Posts/PostsPageSettingsViewController.swift:
```diff
@@ -127,6 +139,73 @@ final class PostsPageSettingsViewController: ViewController, UIPopoverPresentati
             .store(in: &cancellables)
+
+        $immersionModeEnabled
+            .receive(on: RunLoop.main)
```
This will delay updates while scrolling until after scrolling completes, is that intentional? It kinda makes sense to me so I'm ok with it, but a comment would be good. (Usually receiving on the main run loop is not what people want.)

dfsm replied:
I double checked with Claude and it seemed pretty sure that this was preferable. So it added a comment as suggested

nolanw commented on App/View Controllers/Posts/PostsPageSettingsViewController.swift:
```diff
@@ -127,6 +139,73 @@ final class PostsPageSettingsViewController: ViewController, UIPopoverPresentati
+                }
+            }
+            return nil
+        }
```
I don't get it, why don't we have an outlet or property with the stack view handy?

nolanw commented on App/View Controllers/Posts/PostsPageSettingsViewController.swift:
```diff
@@ -127,6 +139,73 @@ final class PostsPageSettingsViewController: ViewController, UIPopoverPresentati
+        // Create immersion mode switch
+        immersionModeSwitch = UISwitch()
+        immersionModeSwitch.isOn = immersionModeEnabled // Set initial state
+        immersionModeSwitch.onTintColor = theme["settingsSwitchColor"] // Apply theme color
```
We should (re)set this in themeDidChange.

nolanw commented on App/View Controllers/Posts/PostsPageView.swift:
```diff
@@ -236,10 +267,14 @@ final class PostsPageView: UIView {
         scrollViewDelegateMux = ScrollViewDelegateMultiplexer(scrollView: renderView.scrollView)
         scrollViewDelegateMux?.addDelegate(self)
+        
+        // Configure initial gradient
```
yep that's what the next line says!

nolanw commented on App/View Controllers/Posts/PostsPageView.swift:
```diff
@@ -251,41 +286,79 @@ final class PostsPageView: UIView {
+            height: gradientHeight)
+        
+        // Update gradient layer frame to match view bounds
+        if let gradientLayer = safeAreaGradientView.layer.sublayers?.first as? CAGradientLayer {
```
This is better done as a UIView subclass that overrides layerClass and does something like
```swift
var gradientLayer: CAGradientLayer { layer as! CAGradientLayer }
```
then we don't have to root around for the gradient layer and set its frame. Also then it works with UIKit animation.

nolanw commented on App/View Controllers/Posts/PostsPageView.swift:
```diff
@@ -327,15 +400,135 @@ final class PostsPageView: UIView {
+        gradientLayer.endPoint = CGPoint(x: 0.5, y: 1.0)
+        
+        safeAreaGradientView.layer.addSublayer(gradientLayer)
+    }
```
All of this could go into that UIView subclass, which could also implement themeDidChange.

nolanw commented on App/View Controllers/Posts/PostsPageView.swift:
```diff
@@ -327,15 +400,135 @@ final class PostsPageView: UIView {
+            self.toolbar.alpha = 1.0
+            self.toolbar.transform = .identity
+            
+            // Hide safe area gradient
```
I recognize these useless comments as very llm coding, which is ok, but I'd love a pass to get rid of them after.

nolanw commented on App/View Controllers/Posts/PostsPageViewController.swift:
```diff
@@ -124,15 +127,35 @@ final class PostsPageViewController: ViewController {
         init() {
             super.init(frame: .zero)
             showsMenuAsPrimaryAction = true
+            // Enable modern iOS 26 menu styling
```
?

nolanw commented on App/View Controllers/Posts/PostsPageViewController.swift:
```diff
@@ -124,15 +127,35 @@ final class PostsPageViewController: ViewController {
+            // Update interface style before showing the menu to ensure it uses the correct theme
+            updateInterfaceStyle()
+            
+            // Use the original approach that was working, but ensure we get iOS 26 styling
```
?

nolanw commented on App/View Controllers/Posts/PostsPageViewController.swift:
```diff
@@ -124,15 +127,35 @@ final class PostsPageViewController: ViewController {
         }
+        
+        func updateInterfaceStyle() {
+            // Follow the theme's mode setting for menu appearance
```
"Menu appearance" is an excellent candidate for a dedicated theme key.

dfsm commented:
So the approach itself has changed from velocity-based to be more of a 1:1 hide/reveal based on user scroll distance.
I removed unnecessary comments (and some were remnants from my fork, sorry)
I added "menuAppearance" to the Themes.plist and gave every theme a value.
Changed the gradient to a UIView and created a new file for it
